src/main/java/sg/ncl/MainController.java

```diff
@@ -794,112 +792,117 @@ public String validateDetails(
-                }
+        log.info("Signup join team success");
+        log.info("jointeam info: {}", joinTeamInfo);
+        redirectAttributes.addFlashAttribute("team", joinTeamInfo);
+        return "redirect:/join_application_submitted";
+    }
 
-                log.info("Signup new team success");
-                return "redirect:/team_application_submitted";
-            }
+    private String checkNewTeamForm(@Valid @ModelAttribute(SIGNUP_MERGED_FORM) SignUpMergedForm signUpMergedForm, RedirectAttributes redirectAttributes, JSONObject mainObject, JSONObject teamFields, String createNewTeamName) {
+        log.info("Signup new team name {}", createNewTeamName);
+        boolean errorsFound = false;
 
-        } else if (joinNewTeamName != null && !joinNewTeamName.isEmpty()) {
-            log.info("Signup join team name {}", joinNewTeamName);
-            // get the team JSON from team name
-            Team2 joinTeamInfo;
+        if (createNewTeamName.length() < 2 || createNewTeamName.length() > 12) {
+            errorsFound = true;
+            signUpMergedForm.setErrorTeamName("Team name must be 2 to 12 alphabetic/numeric characters");
+        }
 
-            try {
-                joinTeamInfo = getTeamIdByName(signUpMergedForm.getJoinTeamName().trim());
-            } catch (TeamNotFoundException | AdapterConnectionException e) {
-                redirectAttributes.addFlashAttribute(MESSAGE, e.getMessage());
-                redirectAttributes.addFlashAttribute("signUpMergedForm", signUpMergedForm);
-                return "redirect:/signup2";
-            }
+        if (signUpMergedForm.getTeamDescription() == null || signUpMergedForm.getTeamDescription().isEmpty()) {
+            errorsFound = true;
+            signUpMergedForm.setErrorTeamDescription("Team description cannot be empty");
+        }
 
-            teamFields.put("id", joinTeamInfo.getId());
+        if (signUpMergedForm.getTeamWebsite() == null || signUpMergedForm.getTeamWebsite().isEmpty()) {
+            errorsFound = true;
+            signUpMergedForm.setErrorTeamWebsite("Team website cannot be empty");
+        }
 
-            // set the flag to indicate to controller that it is joining an existing team
-            mainObject.put("isJoinTeam", true);
+        if (errorsFound) {
+            log.warn("Signup new team error {}", signUpMergedForm.toString());
+            // clear join team name first before submitting the form
+            signUpMergedForm.setJoinTeamName(null);
+            return SIGNUP_PAGE;
+        } else {
+            teamFields.put("name", signUpMergedForm.getTeamName().trim());
+            teamFields.put("description", signUpMergedForm.getTeamDescription().trim());
+            teamFields.put("website", signUpMergedForm.getTeamWebsite().trim());
+            teamFields.put("organisationType", signUpMergedForm.getTeamOrganizationType());
+            teamFields.put("visibility", signUpMergedForm.getIsPublic());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "visibility" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)